### PR TITLE
narwhal: fix narwhal tests and stop primary handler from blocking on deadlock

### PR DESCRIPTION
**Pull request by bmwill**
Turns out that narwhal tests are not blocking today. Lets just merge the CI jobs together so that its easier to ensure they're all blocking.

### CODE REVIEW

**Review by aschran — APPROVED** [review]

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -73,40 +72,16 @@ jobs:
       - uses: taiki-e/install-action@nextest
       - name: cargo test
         run: |
-          cargo nextest run --profile ci -E 'not package(~narwhal)'
+          cargo nextest run --profile ci
+      - name: narwhal celo crypto tests
+        run: cargo nextest run --features celo --profile ci -p narwhal-crypto
       - name: Doctests
         run: |
           cargo test --doc
```
Comment (akichidis) [inline .github/workflows/rust.yml:80]: Let's also ensure that the docs tests will run as `cargo test --doc --features celo`

**Comment by mwtian** [comment]
An alternative as requested a few times is to make the Narwhal CI blocking.

**Comment by bmwill** [comment]
> An alternative as requested a few times is to make the Narwhal CI blocking.

I would strongly prefer merging these jobs in order to dramatically reduce the number of CI minutes spent on doing duplicate builds

**Comment by mwtian** [comment]
With two separate steps, there should be no increase in CI time since building and testing happen in parallel. But I'm totally fine with either way.

**Pull request merged**